### PR TITLE
Add support to disable attributes in ConsoleReporter to be consistent…

### PR DESCRIPTION
**Pull request by husseincoder**
Add support to disable attributes in console reporter to be consistent with GraphiteReporter features .. I saw its easy to be implemented there with not so many changes

### CODE REVIEW

File: metrics-core/src/main/java/com/codahale/metrics/ConsoleReporter.java [hunk 1]
```diff
@@ -317,4 +333,17 @@ private void printWithBanner(String s, char c) {
...
         output.println();
     }
+
+    /**
+     * Print only if the attribute is enabled
+     * @param type Metric attribute
+     * @param status Status to be logged
+     */
+    private void printIfEnabled(MetricAttribute type, String status) {
+        if(isMetricAttributeDisabled(type)) {
```
Comment (arteam) [inline metrics-core/src/main/java/com/codahale/metrics/ConsoleReporter.java]: I think it would better with one condition statement. I also don't think the we need to use `isMetricAttributeDisabled` method as well.
```
if (!getDisabledMetricAttributes().contains(type)) {
    output.println(status);
}
```

File: metrics-core/src/main/java/com/codahale/metrics/ScheduledReporter.java [hunk 2]
```diff
@@ -291,8 +291,17 @@ protected boolean isShutdownExecutorOnStop() {
         return shutdownExecutorOnStop;
     }
 
-    protected Set<MetricAttribute> getDisabledMetricAttributes() {
```
Comment (arteam) [inline metrics-core/src/main/java/com/codahale/metrics/ScheduledReporter.java]: Please, don't remove the `getDisabledMetricAttributes` method. All changes must be backward-compatible.

File: metrics-core/src/main/java/com/codahale/metrics/ScheduledReporter.java [hunk 3]
```diff
@@ -291,8 +291,17 @@ protected boolean isShutdownExecutorOnStop() {
...
-        return disabledMetricAttributes;
+    /**
+     * Check if Attribute disabled to be logged
+     * @param attribute metric attribute
+     * @return True if attribute is configured to be disabled
+     */
+    protected boolean isMetricAttributeDisabled(MetricAttribute attribute) {
```
Comment (arteam) [inline metrics-core/src/main/java/com/codahale/metrics/ScheduledReporter.java]: I don't believe we need to extract the logic of checking of an attribute to a separate method. `disabledMetricAttributes` can't be null, so we basically have one method which does nothing but delegates a call to the `contains` method of `EnumSet`.

File: metrics-core/src/test/java/com/codahale/metrics/ConsoleReporterTest.java [hunk 4]
```diff
@@ -6,10 +6,7 @@
 import java.io.ByteArrayOutputStream;
 import java.io.PrintStream;
 import java.io.UnsupportedEncodingException;
-import java.util.Locale;
```
Comment (arteam) [inline metrics-core/src/test/java/com/codahale/metrics/ConsoleReporterTest.java]: Please, don't introduce star imports if they were not present.

File: metrics-core/src/test/java/com/codahale/metrics/ConsoleReporterTest.java [hunk 5]
```diff
@@ -216,6 +213,168 @@ public void reportsTimerValues() throws Exception {
...
 
+    @Test
+    public void reportMeterWithDisabledAttributes() throws Exception {
+        Set<MetricAttribute> disabledMetricAttributes = EnumSet.of(MetricAttribute.M15_RATE, MetricAttribute.M5_RATE, MetricAttribute.COUNT);
+
+        final ConsoleReporter customReporter = ConsoleReporter.forRegistry(registry)
+                .outputTo(output)
+                .formattedFor(Locale.US)
+                .withClock(clock)
+                .formattedFor(TimeZone.getTimeZone("PST"))
```
Comment (arteam) [inline metrics-core/src/test/java/com/codahale/metrics/ConsoleReporterTest.java]: I guess we could extract the timezone to a variable, so all tests use the same timezone.

File: metrics-core/src/test/java/com/codahale/metrics/ConsoleReporterTest.java [hunk 6]
```diff
@@ -216,6 +213,168 @@ public void reportsTimerValues() throws Exception {
                 ));
     }
 
+    @Test
+    public void reportMeterWithDisabledAttributes() throws Exception {
+        Set<MetricAttribute> disabledMetricAttributes = EnumSet.of(MetricAttribute.M15_RATE, MetricAttribute.M5_RATE, MetricAttribute.COUNT);
+
+        final ConsoleReporter customReporter = ConsoleReporter.forRegistry(registry)
+                .outputTo(output)
+                .formattedFor(Locale.US)
```
Comment (arteam) [inline metrics-core/src/test/java/com/codahale/metrics/ConsoleReporterTest.java]: We could extract the locale to a variable, so all tests use the same locale.

File: metrics-ganglia/src/main/java/com/codahale/metrics/ganglia/GangliaReporter.java [hunk 7]
```diff
@@ -360,7 +360,7 @@ private void reportGauge(String name, Gauge gauge) {
 
     private void announceIfEnabled(MetricAttribute metricAttribute, String metricName, String group, double value, String units)
             throws GangliaException {
-        if (getDisabledMetricAttributes().contains(metricAttribute)) {
+        if (isMetricAttributeDisabled(metricAttribute)) {
```
Comment (arteam) [inline metrics-ganglia/src/main/java/com/codahale/metrics/ganglia/GangliaReporter.java]: Please, don't introduce unrelated changes. Let's focus on `ConsoleReporter`.

File: metrics-graphite/src/main/java/com/codahale/metrics/graphite/GraphiteReporter.java [hunk 8]
```diff
@@ -322,14 +322,14 @@ private void reportHistogram(String name, Histogram histogram, long timestamp) t
     }
 
     private void sendIfEnabled(MetricAttribute type, String name, double value, long timestamp) throws IOException {
-        if (getDisabledMetricAttributes().contains(type)){
+        if (isMetricAttributeDisabled(type)){
```
Comment (arteam) [inline metrics-graphite/src/main/java/com/codahale/metrics/graphite/GraphiteReporter.java]: Let's avoid changes in the classes not directly targeted by this pull request.

**Comment by husseincoder** [comment]
@arteam Thanks for the code review .. I've changed these stuff and gonna push them soon

**Comment by arteam** [comment]
Thanks for your hard work on this, looks great!
